Allow preview release version during NuGet publishing (OSOE-578)

Dose the release version number  must be numeric only?

![image](https://user-images.githubusercontent.com/15613121/220831876-4def53c6-93c3-4d9b-9a42-b842189a4056.png)
 
```
Error: "v1.0.9-beta" is invalid. The correct format is "vM.N.O" or "vM.N.O-alpha.X.<issue-code>".
Error: Process completed with exit code 1.
``` 

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-578)

Error: "v1.0.9-alpha.0" is invalid. The correct format is "vM.N.O" or "vM.N.O-alpha.X.<issue-code>".
Error: Process completed with exit code 1.
 
![image](https://user-images.githubusercontent.com/15613121/220865020-86ef163e-e7cf-4eec-99c5-af2f8531d2d0.png)
 
> you can reference an earlier commit hash of this repo in your workflow rather than @dev to get the old behavior.

@Piedone  Sorry, I don't understand what you mean, you mean I copy this script to my repository, right?

--- a/.github/actions/publish-nuget/action.yml
+++ b/.github/actions/publish-nuget/action.yml
@@ -62,8 +62,8 @@ runs:
 
         if ($version -eq 'USE_GITHUB_REF_NAME') {
             $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v")
-            if (!($version -match '^\d+\.\d+\.\d+(-alpha\.\d+\.[a-z]{2,}-\d+)?$')) {
-                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-alpha.X.<issue-code>`"."
+            if (!($version -match '^\d+\.\d+\.\d+(-(beta|preview|alpha|rc)(\.\d+(\.[a-zA-Z]{2,}(-\d+)?)?)?)?$')) {
+                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-[alpha, preview , beta , rc].X.<issue-code>`"."
                 exit 1
             }
         }

--- a/.github/actions/publish-nuget/action.yml
+++ b/.github/actions/publish-nuget/action.yml
@@ -62,8 +62,8 @@ runs:
 
         if ($version -eq 'USE_GITHUB_REF_NAME') {
             $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v")
-            if (!($version -match '^\d+\.\d+\.\d+(-(beta|preview|alpha|rc)(\.\d+(\.[a-zA-Z]{2,}(-\d+)?)?)?)?$')) {
-                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-[alpha, preview , beta , rc].X.<issue-code>`"."
+            if (!($version -match '^\d+\.\d+\.\d+(-(alpha|beta|preview|rc)(\.\d+(\.[a-zA-Z]{2,}(-\d+)?)?)?)?$')) {
+                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-[alpha|beta|preview|rc]` with an optional `.X.<issue-code>` suffix"."
                 exit 1
             }
         }

